src/controllers/user/official-contact-email-verification.ts

```diff
@@ -105,24 +106,22 @@ export const postOfficialContactEmailVerificationMiddleware = async (
       params: req.params,
     });
 
-    await verifyOfficialContactEmailToken({
-      user_id: getUserFromAuthenticatedSession(req).id,
-      organization_id,
-      token: official_contact_email_verification_token,
-    });
+    const [error] = await to(
+      verifyOfficialContactEmailToken({
+        user_id: getUserFromAuthenticatedSession(req).id,
+        organization_id,
+        token: official_contact_email_verification_token,
+      }),
+    );
 
-    return next();
-  } catch (error) {
-    const { organization_id } = req.params;
-    if (
-      organization_id &&
-      (error instanceof InvalidTokenError || error instanceof ZodError)
-    ) {
+    if (error instanceof InvalidTokenError || error instanceof ZodError) {
```

Comment:
Error should not be ZodError here.

ZodError are passed to catch has it should be.

Reply:
see https://github.com/numerique-gouv/moncomptepro/pull/675/files